src/service.rs

```diff
@@ -390,15 +388,12 @@ where
                         continue;
                     }
                     Ok(recv_attempt) => {
-                        if recv_attempt.is_ok() {
-                            break recv_attempt.map_err(|_| BuildError::Missing);
-                        } else {
-                            let e = recv_attempt.unwrap_err();
+                        if let Err(e) = recv_attempt {
                             tracing::error!(%e, "Channel closed while getting available blocks for parent {:?}", req_msg.requested_vid_commitment);
-                            break Err(BuildError::Error {
-                                message: "channel unexpectedly closed".to_string(),
-                            });
                         }
+                        break recv_attempt.map_err(|_| BuildError::Error {
```

Comment:
This is better but still confusing IMO. We check for an error, log it, but then fall through and handle the error again with `map_err`? What about collapsing this into one error handling branch, like
```rust
break recv_attempt.map_err(|e| {
  tracing::error!(%e, "Channel closed blah blah blah");
  BuildError::Error {
    message: "blah"
  }
});
```

Reply:
Yeah, I thought that would be more confusing, but it's fine that way too.